test/blackbox-tests/test-cases/pkg/ocamlformat/ocamlformat-patch-extra-files.t

```diff
@@ -0,0 +1,83 @@
+'dune fmt' produce lock files inside "dev-tools.locks/ocamlformat" before it starts building
+ocamlformat and its dependencies during the same run of the "dune fmt" command. The source tree is
+loaded before the lock files are produced, this is why any 'patch' file inside
+'dev-tools.locks/ocmalformat' is not copied inside the 'build' directory when a rule depends on it.
+
+In the case of this issue, there is a rule that depends on an 'patch' file in order to copy the file
+inside '_private/default/..' directory, since the file could not be copied, the rule is not activated.
+So it fails when 'dune' trying to apply the patch. After any follwing run of 'dune fmt', it works
+because the 'patch' file is already present.
+
+  $ . ./helpers.sh
+  $ mkrepo
+
+Make a fake ocamlformat:
+  $ make_fake_ocamlformat "0.1"
+
+  $ cat > patch-for-ocamlformat.patch <<EOF
+  > diff a/ocamlformat.ml b/ocamlformat.ml
+  > --- a/ocamlformat.ml
+  > +++ b/ocamlformat.ml
+  > @@ -1,6 +1,6 @@
+  > -let version = "0.1"
+  > +let version = "0.26.2"
+  >  let () =
+  >    if Sys.file_exists ".ocamlformat-ignore" then
+  >    print_endline "ignoring some files"
+  >  ;;
+  >  let () = print_endline ("formatted with version "^version)
+  > EOF
+
+Make the ocamlformat opam package which uses a patch:
+  $ mkpkg ocamlformat "0.26.2"  <<EOF
```

Comment:
Do you actually need the solving step? Can't you write a fake lock directory for ocamlformat?

Reply:
Yeah we need it. The bug is when the solving step is done during the fisrt run of  `dune fmt` and if you have already the lock directory it won't trigger the bug.